### ISSUE TITLE
[Tag] List of tags cause hidden focus styles

### ISSUE THREAD

**Issue by AndrewMusgrave** [Shopify/polaris#3656]
## Issue summary

Tag use overflow/text-overflow to manage tag content. However, this doesn't work well in lists of tags. The consumer can apply `overflow: hidden` to the list items, however, this breaks the focus styles.

## Expected behavior

* Visible focus styles

## Reduced test case

[Code sandbox](https://codesandbox.io/s/romantic-khayyam-fdyti?file=/App.js)
